libtenzir/builtins/functions/int.cpp

```diff
@@ -47,6 +47,32 @@ class int_uint final : public function_plugin {
         [](const Array& arg) {
           return std::make_shared<Array>(arg.data());
         },
+        [&]<class T>(const T& arg)
+          requires integral_type<type_from_arrow_t<T>>
```

Comment:
Do we want this as a concept if it's used in more places?

Reply:
Probably not worth it right now, but if you specify it generally then you'd need to restrict it further like this:

```
template <class T>
concept integral_array = integral_type<type_from_arrow_t<T>> && std::is_base_of_v<arrow::Array, T>;
```